olp-cpp-sdk-core/tests/thread/ContinuationTest.cpp

```diff
@@ -87,21 +87,24 @@ TEST_F(ContinuationTest, CancelBeforeRun) {
 }
 
 TEST_F(ContinuationTest, CancelAfterRun) {
-  std::promise<olp::client::ApiError> promise;
+  std::promise<Response<int>> promise;
   auto future = promise.get_future();
 
   auto continuation = Create([](olp::thread::ExecutionContext,
                                 std::function<void(int)> next) { next(3); })
                           .Then([](olp::thread::ExecutionContext, int,
                                    std::function<void(int)> next) { next(3); })
                           .Finally([&promise](Response<int> response) {
-                            promise.set_value(response.GetError());
+                            promise.set_value(response.MoveResult());
                           });
 
   continuation.Run();
+  // simulate a delay to make sure the continuation is finished.
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
   continuation.CancelToken().Cancel();
 
-  EXPECT_EQ(future.get().GetErrorCode(), olp::client::ErrorCode::Cancelled);
+  auto result = future.get();
+  EXPECT_TRUE(result.IsSuccessful());
```

Comment:
```suggestion
  EXPECT_TRUE(result);
```

Please do the same for all similar cases

Reply:
done